### ISSUE TITLE
syntax highlighting error when rendering submodule diff

### ISSUE THREAD

**Issue by shiftkey** [desktop/desktop#7691]
## Description

Identified while looking into a React issue, but I don't believe this is a regression in 2.0

## Version
* GitHub Desktop: 2.0.0
* Operating system: macOS 10.14.5

## Steps to Reproduce
1. Move a submodule inside a repository to a different commit
2. View the working directory changes

<img width="790" src="https://user-images.githubusercontent.com/359239/58961999-9fea3200-8780-11e9-871b-55b21e4baddc.png">

3. Check the DevTools for errors

### Expected Behavior
Syntax highlighting worker process detects and gracefully avoids rendering subnmodule diff

### Actual Behavior

Two errors logged each time it tries to update the diff:

<img width="1289"  src="https://user-images.githubusercontent.com/359239/58962115-e17add00-8780-11e9-9f1a-8b1bb0646666.png">

```
Could not load new contents for syntax highlighting
Error: EISDIR: illegal operation on a directory, read
```

```
Could not load old contents for syntax highlighting
Error: Git returned an unexpected exit code '128' which should be handled by the caller.'
    at ChildProcess.s.on (/webpack:/app/src/lib/git/spawn.ts:93:15)
    at ChildProcess.emit (events.js:182:13)
    at maybeClose (internal/child_process.js:961:16)
    at Socket.stream.socket.on (internal/child_process.js:380:11)
    at Socket.emit (events.js:182:13)
    at Pipe._handle.close [as _onclose] (net.js:596:12)
```

**Comment by tierninho** [comment]
Another example of this error from the wild here: https://github.com/desktop/desktop/issues/8064

**Comment by sergiou87** [comment]
Partially fixed with #15176 

However, that PR only applies to the changes list, not a commit diff.
